### ISSUE TITLE
Rate limit exceeded

### ISSUE THREAD

**Issue by jazzdan** [benbrittain/buckle#22]
I was doing development normally today, invoking ~10 builds/hour using buckle, and eventually Buckle broke:

```
$ buckle
Error: invalid type: map, expected a sequence at line 1 column 0
```

I looked in the releases.json and saw that it contained an error message:

```
$ cat /Users/dan/Library/Caches/buckle/releases.json
{"message":"API rate limit exceeded for a.b.c.d. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
```

It seems like buckle should definitely not overwrite releases.json with an error message if it gets rate limited, and I'm surprised that I was rate limited at all given how lightly I was using it.

**Comment by benbrittain** [comment]
Yeah, that's no good at all. I haven't seen this occur personally but I'll definitely improve that logic

**Comment by rbtcollins** [comment]
Using the GitHub API without a token will get somewhat arbitrary rate limits applied - I was seeing nearly a 25% failure rate in GHA earlier today for instance. I suggest permitting buckle to accept a GITHUB_TOKEN variable or option and use that when looking up releases.
